test/functional/apps/dashboard/group5/share.ts

```diff
@@ -1,167 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0 and the Server Side Public License, v 1; you may not use this file except
- * in compliance with, at your election, the Elastic License 2.0 or the Server
- * Side Public License, v 1.
- */
-
-import expect from '@kbn/expect';
-import { FtrProviderContext } from '../../../ftr_provider_context';
-
-type TestingModes = 'snapshot' | 'savedObject';
-type AppState = string | undefined;
-interface UrlState {
-  globalState: string;
-  appState: AppState;
-}
-
-const getStateFromUrl = (url: string): UrlState => {
-  const globalStateStart = url.indexOf('_g');
-  const appStateStart = url.indexOf('_a');
-
-  // global state is always part of the URL, but app state is *not* - so, need to
-  // modify the logic depending on whether app state exists or not
-  if (appStateStart === -1) {
-    return {
-      globalState: url.substring(globalStateStart + 3),
-      appState: undefined,
-    };
-  }
-  return {
-    globalState: url.substring(globalStateStart + 3, appStateStart - 1),
-    appState: url.substring(appStateStart + 3, url.length),
-  };
-};
-
-export default function ({ getService, getPageObjects }: FtrProviderContext) {
-  const retry = getService('retry');
-  const filterBar = getService('filterBar');
-  const kibanaServer = getService('kibanaServer');
-  const testSubjects = getService('testSubjects');
-  const dashboardPanelActions = getService('dashboardPanelActions');
-  const dashboardCustomizePanel = getService('dashboardCustomizePanel');
-
-  const PageObjects = getPageObjects(['dashboard', 'common', 'share', 'timePicker']);
-
-  const getSharedUrl = async (mode: TestingModes): Promise<string> => {
-    await retry.waitFor('share menu to open', async () => {
-      await PageObjects.share.clickShareTopNavButton();
-      return await PageObjects.share.isShareMenuOpen();
-    });
-    if (mode === 'savedObject') {
-      await PageObjects.share.exportAsSavedObject();
-    }
-    return PageObjects.share.getSharedUrl();
-  };
```

Comment:
I'm not convinced that these tests are redundant - we might just have to modify them to check the short URL saved object instead of the URL directly... I still feel like ensuring that the generated URL looks as expected / includes the expected part is important, and it now feels like we aren't testing that at all. 

How do you feel about this, @rshen91? :eyes:

Reply:
I have an issue open for this here! https://github.com/elastic/kibana/issues/181066